File: Implementações/BubbleSort.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>

// Função para trocar dois elementos
void swap(int *a, int *b)
{
  int temp = *a;
  *a = *b;
  *b = temp;
}

// Implementação do Bubble Sort
void bubbleSort(int arr[], int n)
{
  for (int i = 0; i < n - 1; i++)
  {
    for (int j = 0; j < n - i - 1; j++)
    {
      if (arr[j] > arr[j + 1])
      {
        swap(&arr[j], &arr[j + 1]);
      }
    }
  }
}

// Função para gerar dados aleatórios
void generateRandomArray(int arr[], int n)
{
  for (int i = 0; i < n; i++)
  {
    arr[i] = rand() % 10000; // Números entre 0 e 9999
  }
}

// Função para exibir o array
void printArray(int arr[], int n)
{
  for (int i = 0; i < n; i++)
  {
    printf("%d ", arr[i]);
  }
  printf("\n");
}

int main()
{
  int n = 10000; // Tamanho do array (você pode alterar para testar diferentes tamanhos)
  int arr[n];

  // Gerar array aleatório
  generateRandomArray(arr, n);

  printf("Array original:\n");
  printArray(arr, n);

  // Medir o tempo de execução
  clock_t start, end;
  start = clock();
  bubbleSort(arr, n);
  end = clock();

  printf("Array ordenado:\n");
  printArray(arr, n);

  double time_taken = ((double)(end - start)) / CLOCKS_PER_SEC;
  printf("Tempo de execucao: %f segundos\n", time_taken);

  return 0;
}

File: ImplementaçõesPython/BubbleSort.py
import time
import random

# Implementação do Bubble Sort
def bubble_sort(arr):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

# Função para gerar dados aleatórios
def gerar_array_aleatorio(n, max_value=10000):
    return [random.randint(0, max_value) for _ in range(n)]

# Main
if __name__ == "__main__":
    n = 10000  # Tamanho do array (mude para testar diferentes tamanhos)
    arr = gerar_array_aleatorio(n)

    print("Array original:")
    print(arr)

    # Medir o tempo de execução
    inicio_tempo = time.time()
    bubble_sort(arr)
    fim_tempo = time.time()

    print("Array ordenado:")
    print(arr)

    print(f"Tempo de execução: {fim_tempo - inicio_tempo:.6f} segundos")
